skymarshal/skycmd/github_flags.go

```diff
@@ -61,13 +63,21 @@ func (flag *GithubFlags) Serialize(redirectURI string) ([]byte, error) {
 type GithubTeamFlags struct {
 	Users []string `long:"user" description:"List of whitelisted GitHub users" value-name:"USERNAME"`
 	Orgs  []string `long:"org" description:"List of whitelisted GitHub orgs" value-name:"ORG_NAME"`
-	Teams []string `long:"team" description:"List of whitelisted GitHub teams" value-name:"ORG_NAME:TEAM_NAME"`
+	Teams []string `long:"team" description:"List of whitelisted GitHub teams. TEAM_NAME could be github team name or team slug" value-name:"ORG_NAME:TEAM_NAME"`
```

Comment:
@vito I like the idea of typed value, which seems much cleaner. Just one thing though, `fly set-team` with config file needs to change accordingly.

Say if the type is
```
type GithubTeamFlag struct {
	Org  string `yaml:"org"`
	Team string `yaml:"team"`
}
```

Then the config yaml will change from
```
...
  - name: viewer
    github:
      teams: ["some-org:some-team"]
      users: ["some-github-user"]
...
```
to
```
...
  - name: viewer
    github:
      teams:
        - org: "some-org"
          team: "some-team"
      users: ["some-github-user"]
...
```
because of this line https://github.com/concourse/concourse/blob/75e7d95e1d521bd6bcff7bcb781c64aff64a0690/skymarshal/skycmd/flags.go#L110

Just want to make sure on this.

Reply:
@vito I like the idea. Please kindly correct me if i am wrong but it only solve half the problem since `fly set-team` also takes config yaml file.

Do we need to use decoder hook here https://github.com/concourse/concourse/blob/75e7d95e1d521bd6bcff7bcb781c64aff64a0690/skymarshal/skycmd/flags.go#L110
to parse the value of the type too? If it is true than I am not 100% about that.

Maybe it isn't worth the effort?

Reply:
Dang, yeah probably not worth it for now. I'd really like to stop using `mapstructure` because of this - it's way too annoying to write an unmarshaler for it. I've got #3505 in flight for that for pipeline YAMLs but haven't looked at team configs yet.